engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/immutable/RemoveAtVectorNode.java

```diff
@@ -27,69 +19,19 @@ static RemoveAtVectorNode build() {
     return RemoveAtVectorNodeGen.create();
   }
 
-  abstract Vector execute(Object vec, long index);
+  abstract EnsoObject execute(Object vec, long index);
 
   @Specialization
-  Vector fromVector(
-      Vector vec,
+  EnsoObject removeAtIndex(
+      Object storage,
       long index,
-      @Shared("copyNode") @Cached CopyNode copyNode,
-      @Shared("interop") @CachedLibrary(limit = "3") InteropLibrary interop) {
-    try {
-      return removeAtIndex(vec.toArray(), index, copyNode, interop);
-    } catch (UnsupportedMessageException e) {
-      CompilerDirectives.transferToInterpreter();
-      Builtins builtins = EnsoContext.get(this).getBuiltins();
-      throw new PanicException(builtins.error().makeTypeError(builtins.vector(), vec, "vec"), this);
-    }
-  }
-
-  @Specialization
-  Vector fromArray(
-      Array vec,
-      long index,
-      @Shared("copyNode") @Cached CopyNode copyNode,
-      @Shared("interop") @CachedLibrary(limit = "3") InteropLibrary interop) {
-    try {
-      return removeAtIndex(vec, index, copyNode, interop);
-    } catch (UnsupportedMessageException e) {
-      throw unsupportedException(vec);
-    }
-  }
-
-  @Specialization(guards = "interop.hasArrayElements(vec)")
-  Vector fromArrayLike(
-      Object vec,
-      long index,
-      @Shared("copyNode") @Cached CopyNode copyNode,
-      @Shared("interop") @CachedLibrary(limit = "3") InteropLibrary interop) {
-    try {
-      return removeAtIndex(vec, index, copyNode, interop);
-    } catch (UnsupportedMessageException e) {
-      throw unsupportedException(vec);
-    }
-  }
-
-  @Fallback
-  Vector fromUnknown(Object vec, long index) {
-    throw unsupportedException(vec);
-  }
-
-  private PanicException unsupportedException(Object vec) {
-    CompilerDirectives.transferToInterpreter();
-    var ctx = EnsoContext.get(this);
-    var err = ctx.getBuiltins().error().makeTypeError("polyglot array", vec, "vec");
-    throw new PanicException(err, this);
-  }
-
-  private Vector removeAtIndex(
-      Object storage, long index, CopyNode copyArrayNode, InteropLibrary interop)
-      throws UnsupportedMessageException {
-    long length = interop.getArraySize(storage);
+      @Cached ArrayLikeCopyToArrayNode copyArrayNode,
+      @Cached ArrayLikeLengthNode lengthNode) {
+    long length = lengthNode.executeLength(storage);
     long actualIndex = index < 0 ? index + length : index;
-    Array array = Array.allocate(length - 1);
+    var array = ArrayLikeHelpers.allocate(length - 1);
     copyArrayNode.execute(storage, 0, array, 0, actualIndex);
     copyArrayNode.execute(storage, actualIndex + 1, array, actualIndex, length - actualIndex - 1);
-    return Vector.fromArray(array);
+    return ArrayLikeHelpers.asVectorFromArray(array);
   }
```

Comment:
Add a fallback that just throws a `Type_Error`? It used to be here before.

Reply:
`PanicException` is going to be raised from `ArrayLikeAtNode`. Some cleanup in e801896d0e